lib/network/http/utils.js

```diff
@@ -0,0 +1,109 @@
+'use strict'; // eslint-disable-line
+
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into an object
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+
+ * @return {Object} object containing a range specification, with
+ * either of:
+ * - start and end attributes: a fully specified range request
+ * - a single start attribute: no end is specified in the range request
+ * - a suffix attribute: suffix range request
+ * - an error attribute of type errors.InvalidArgument if the range
```

Comment:
these are the tests I am referring to: https://github.com/ceph/s3-tests
See integration for how to run them: https://github.com/scality/Integration/blob/master/genericStaas/env/run_cephtests.bash